File: server/tests/fixtures/alpha_rule_triggers/csharp/complexity_high_cyclomatic.cs
// Should trigger: complexity.high_cyclomatic
public class HighComplexity {
    public string Classify(int a, int b, int c, int d, int e) {
        if (a > 0) {
            if (b > 0) {
                if (c > 0) {
                    if (d > 0) {
                        if (e > 0) {
                            return "all positive";
                        } else if (e < 0) {
                            return "e negative";
                        }
                    } else if (d < 0) {
                        return "d negative";
                    }
                } else if (c < 0) {
                    return "c negative";
                }
            } else if (b < 0) {
                return "b negative";
            }
        } else if (a < 0) {
            return "a negative";
        }
        return "default";
    }
}

File: server/tests/fixtures/alpha_rule_triggers/typescript/complexity_high_cyclomatic.ts
// Should trigger: complexity.high_cyclomatic
function complexFunction(a: boolean, b: boolean, c: boolean, d: boolean, e: boolean, f: boolean, g: boolean, h: boolean, i: boolean, j: boolean, k: boolean): number {
    if (a) return 1;
    else if (b) return 2;
    else if (c) return 3;
    else if (d) return 4;
    else if (e) return 5;
    else if (f) return 6;
    else if (g) return 7;
    else if (h) return 8;
    else if (i) return 9;
    else if (j) return 10;
    else if (k) return 11;
    else return 0;
}
